lib/queuePopulator/QueuePopulator.js

```diff
@@ -417,6 +417,8 @@ class QueuePopulator {
             try {
                 this.bnConfigManager = new NotificationConfigManager({
                     mongoConfig: this.qpConfig.mongo,
+                    bucketMetastore: this.extConfigs.notification.bucketMetastore,
+                    zkClient: this.qpConfig.mongo ? null : this.zkClient,
```

Comment:
why filter the parameters?
NotificationConfigManager takes multiple optional params (mongo, zkClient...) and deals with it as needed, no point doing it for him here.